### ISSUE TITLE
Failed to parse warning

### ISSUE THREAD

**Issue by maennchen** [jeremyjh/dialyxir#380]
### Precheck

* Take a look at the open issues and be sure that your issue is not already covered.
* Be sure your versions of Dialyxir and Erlex are up to date.

### Environment

* Elixir & Erlang/OTP versions (elixir --version):

```
Erlang/OTP 22 [erts-10.4] [source] [64-bit] [smp:6:6] [ds:6:6:10] [async-threads:1] [hipe]

Elixir 1.9.4 (compiled with Erlang/OTP 22)
```

* Which version of Dialyxir are you using? (cat mix.lock | grep dialyxir):

```
  "dialyxir": {:hex, :dialyxir, "1.0.0-rc.7", "6287f8f2cb45df8584317a4be1075b8c9b8a69de8eeb82b4d9e6c761cf2664cd", [:mix], [{:erlex, ">= 0.2.5", [hex: :erlex, repo: "hexpm", optional: false]}], "hexpm"},
```

Used Libraries? (cat mix.lock | grep cldr)

```
  "cldr_utils": {:hex, :cldr_utils, "2.5.0", "2a15b82b5b56bba99b897ff5801a5b1dcbce425b6430445e97d024a9999afb03", [:mix], [{:decimal, "~> 1.5", [hex: :decimal, repo: "hexpm", optional: false]}], "hexpm"},
  "ex_cldr": {:hex, :ex_cldr, "2.12.0", "929204d74537df28147a22b77acf054cb9fee4b11a1dbffbaa6b522750d55396", [:mix], [{:cldr_utils, "~> 2.3", [hex: :cldr_utils, repo: "hexpm", optional: false]}, {:decimal, "~> 1.6", [hex: :decimal, repo: "hexpm", optional: false]}, {:gettext, "~> 0.13", [hex: :gettext, repo: "hexpm", optional: true]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}, {:nimble_parsec, "~> 0.5", [hex: :nimble_parsec, repo: "hexpm", optional: false]}, {:plug, "~> 1.4", [hex: :plug, repo: "hexpm", optional: true]}, {:telemetry, "~> 0.4.0", [hex: :telemetry, repo: "hexpm", optional: true]}], "hexpm"},
  "ex_cldr_calendars": {:hex, :ex_cldr_calendars, "1.6.0", "ada9bb551fbd76325199094b06a83fac3949c1d2022d05dd6efbe8046fbcd8fa", [:mix], [{:ex_cldr, "~> 2.8", [hex: :ex_cldr, repo: "hexpm", optional: false]}, {:ex_cldr_lists, "~> 2.4", [hex: :ex_cldr_lists, repo: "hexpm", optional: true]}, {:ex_cldr_units, "~> 2.0", [hex: :ex_cldr_units, repo: "hexpm", optional: true]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: false]}], "hexpm"},
  "ex_cldr_currencies": {:hex, :ex_cldr_currencies, "2.4.1", "a8e8330a6d0712b8bb34c5e3759311da1d53fa8bebef163d72615f0ea60c0738", [:mix], [{:ex_cldr, "~> 2.6", [hex: :ex_cldr, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}], "hexpm"},
  "ex_cldr_dates_times": {:hex, :ex_cldr_dates_times, "2.2.3", "4a82f6af48f55c92c0d28be066e5b2451e807e3f49246d08845af664fd7cb712", [:mix], [{:ex_cldr, "~> 2.8", [hex: :ex_cldr, repo: "hexpm", optional: false]}, {:ex_cldr_calendars, "~> 1.2", [hex: :ex_cldr_calendars, repo: "hexpm", optional: false]}, {:ex_cldr_numbers, "~> 2.6", [hex: :ex_cldr_numbers, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}], "hexpm"},
  "ex_cldr_lists": {:hex, :ex_cldr_lists, "2.4.0", "08e0857f93bdfde45905934475effc1d517cf58a67a6e676837d16f5f2fd85e1", [:mix], [{:ex_cldr, "~> 2.8", [hex: :ex_cldr, repo: "hexpm", optional: false]}, {:ex_cldr_numbers, "~> 2.6", [hex: :ex_cldr_numbers, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}], "hexpm"},
  "ex_cldr_numbers": {:hex, :ex_cldr_numbers, "2.9.0", "ffba3ba8cfa41194e57cc38f0cc78f83956ce5c1822b1e4d5f64394cb927ba2c", [:mix], [{:decimal, "~> 1.5", [hex: :decimal, repo: "hexpm", optional: false]}, {:ex_cldr, "~> 2.11", [hex: :ex_cldr, repo: "hexpm", optional: false]}, {:ex_cldr_currencies, "~> 2.3", [hex: :ex_cldr_currencies, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}], "hexpm"},
  "ex_cldr_units": {:hex, :ex_cldr_units, "2.7.0", "c7f87f21c07b09c241e39b0446be38c7036a7ed228ad0485cb12db2dbc9630e9", [:mix], [{:ex_cldr, "~> 2.11", [hex: :ex_cldr, repo: "hexpm", optional: false]}, {:ex_cldr_lists, "~> 2.2", [hex: :ex_cldr_lists, repo: "hexpm", optional: false]}, {:ex_cldr_numbers, "~> 2.6", [hex: :ex_cldr_numbers, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}], "hexpm"},
  "ex_money": {:hex, :ex_money, "4.4.2", "7c35e05eda5efeedf1bbe263bf097b9c240e02f73c5d56e1bd69467f3e625e5b", [:mix], [{:decimal, "~> 1.6", [hex: :decimal, repo: "hexpm", optional: false]}, {:ex_cldr, "~> 2.12", [hex: :ex_cldr, repo: "hexpm", optional: false]}, {:ex_cldr_currencies, "~> 2.4", [hex: :ex_cldr_currencies, repo: "hexpm", optional: false]}, {:ex_cldr_numbers, "~> 2.6", [hex: :ex_cldr_numbers, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}, {:nimble_parsec, "~> 0.5", [hex: :nimble_parsec, repo: "hexpm", optional: true]}, {:phoenix_html, "~> 2.0", [hex: :phoenix_html, repo: "hexpm", optional: true]}], "hexpm"},
```

### Current behavior

```
$ mix dialyzer
In an Umbrella child, not checking PLT...
No :ignore_warnings opt specified in mix.exs and default does not exist.

Starting Dialyzer
[
  check_plt: false,
  init_plt: '/home/maennchen/Development/.../_build/dev/dialyxir_erlang-22.0_elixir-1.9.4_deps-dev.plt',
  files: [...],
  warnings: [:unknown]
]
Total errors: 5, Skipped: 0, Unnecessary Skips: 0
done in 0m11.22s
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to parse warning:
[{:"<<", 1}, {:<, 1}, {:_, 1}, {:":", 1}, {:int, 1, 8}, {:",", 1}, {:_, 1}, {:":", 1}, {:_, 1}, {:*, 1}, {:int, 1, 8}, {:">>", 1}, {:",", 1}, {:_, 1}, {:>, 1}]


Legacy warning:
lib/cldr/backend/formatter.ex:1: The pattern <#{'number_system':=__@1}, __@2> can never match the type <<<_:8,_:_*8>>,_>
________________________________________________________________________________
...
________________________________________________________________________________
done (warnings were emitted)
Halting VM with exit status 2
```

### Expected behavior

A readable error.

**Comment by maennchen** [comment]
Oh, I already did some tinkering, the original error was:

```
$ mix dialyzer
In an Umbrella child, not checking PLT...
No :ignore_warnings opt specified in mix.exs and default does not exist.

Starting Dialyzer
[
  check_plt: false,
  init_plt: '/home/maennchen/Development/.../_build/dev/dialyxir_erlang-22.0_elixir-1.9.4_deps-dev.plt',
  files: [...],
  warnings: [:unknown]
]
** (throw) {:error, :parsing, [{:"<<", 1}, {:<, 1}, {:_, 1}, {:":", 1}, {:int, 1, 8}, {:",", 1}, {:_, 1}, {:":", 1}, {:_, 1}, {:*, 1}, {:int, 1, 8}, {:">>", 1}, {:",", 1}, {:_, 1}, {:>, 1}]}
    lib/erlex.ex:59: Erlex.parse/1
    lib/erlex.ex:85: Erlex.pretty_print/1
    lib/erlex.ex:172: Erlex.pretty_print_type/1
    lib/dialyxir/warnings/pattern_match.ex:11: Dialyxir.Warnings.PatternMatch.format_short/1
    lib/dialyxir/formatter.ex:71: Dialyxir.Formatter.format_warning/2
    lib/dialyxir/formatter.ex:209: Dialyxir.Formatter.filter_warning/3
    (elixir) lib/enum.ex:1440: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    lib/dialyxir/formatter.ex:199: Dialyxir.Formatter.filter_warnings/3
```

Change from my side:

Added this code to `Dialyxir.Warnings.PatternMatch.format_short/1`.

```elixir
catch
  {:error, :parsing, _} -> "The pattern can never match the erlang type #{type}."
```

**Comment by kipcole9** [comment]
Here is a minimal repo to reproduce: https://github.com/kipcole9/dt_format_test

**Comment by maennchen** [comment]
@kipcole9 Thanks, that solves the issue :)

**Comment by asummers** [comment]
This is fixed in Erlex 0.2.6
